### PR TITLE
Fixing HPA not converting apiGroup for deploymentconfigs

### PR DESCRIPTION
**Pull request by ammachado**
ApiGroup for HPA ScaleTargetRef when it is a deploymentconfig is not changed to the new version.

### CODE REVIEW

File: velero-plugins/horizontalpodautoscaler/restore.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+			var out map[string]interface{}
+			objrec, _ := json.Marshal(hpa)
+			json.Unmarshal(objrec, &out)
+
+			return velero.NewRestoreItemActionExecuteOutput(&unstructured.Unstructured{Object: out}), nil
+		}
+	}
+
+	p.Log.Info("[hpa-restore] Route has statically-defined host so leaving as-is")
```
Comment (pranavgaikwad) [inline velero-plugins/horizontalpodautoscaler/restore.go]:
```suggestion
	p.Log.Info("[hpa-restore] ScaleTargetRef not a DeploymentConfig, leaving as-is")
```

File: velero-plugins/horizontalpodautoscaler/restore.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+// AppliesTo returns a velero.ResourceSelector that applies to everything
+func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"horizontalpodautoscalers"},
+	}, nil
+}
+
+// Execute fixes the route path on restore to use the target cluster's domain name
```
Comment (pranavgaikwad) [inline velero-plugins/horizontalpodautoscaler/restore.go]:
```suggestion
// Execute fixes apiVersion in ScaleTargetRef of HPA 
```

File: velero-plugins/horizontalpodautoscaler/restore.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+func (p *RestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput) (*velero.RestoreItemActionExecuteOutput, error) {
+	p.Log.Info("[hpa-restore] Entering HorizontalPodAutoscaler restore plugin")
+	hpa := v2beta1.HorizontalPodAutoscaler{}
+	itemMarshal, _ := json.Marshal(input.Item)
+	json.Unmarshal(itemMarshal, &hpa)
+
+	if (v2beta1.CrossVersionObjectReference{}) != hpa.Spec.ScaleTargetRef {
+		gv, err := schema.ParseGroupVersion(hpa.Spec.ScaleTargetRef.APIVersion)
+		if err != nil {
+			return velero.NewRestoreItemActionExecuteOutput(input.Item), nil
```
Comment (pranavgaikwad) [inline velero-plugins/horizontalpodautoscaler/restore.go]: Shouldn't an error returned in this case? returning nil just fails it silently

**Comment by pranavgaikwad** [comment]
@ammachado Thank you for the fix.

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]

File: velero-plugins/common/restore.go [hunk 4]
```diff
@@ -41,7 +41,8 @@ func (p *RestorePlugin) AppliesTo() (velero.ResourceSelector, error) {
 			"replicasets.extensions",
 			"replicationcontroller",
 			"buildconfigs",
-			"buildconfigs.build.openshift.io"},
+			"buildconfigs.build.openshift.io",
+			"horizontalpodautoscalers"},
```
Comment (sseago) [inline velero-plugins/common/restore.go]: Why is this needed? I don't see anything in the below horizontalpodautoscaler restore plugin that relies on the actions of the common restore plugin here.

Comment (ammachado) [reply velero-plugins/common/restore.go]: I am not sure if it is needed, but on this list is all of the plugins that are present in the application.

Comment (pranavgaikwad) [reply velero-plugins/common/restore.go:45]: @ammachado This can be removed from the common plugin since we have a separate plugin registered for hpa

Comment (pranavgaikwad) [reply velero-plugins/common/restore.go]: @ammachado This can be removed from the common plugin since we have a separate plugin registered for hpa

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]

File: velero-plugins/common/restore.go (hunk 4 above)
Comment (sseago) [reply velero-plugins/common/restore.go]: @ammachado This list the resources that specifically need the actions taken in this plugin. Originally we applied these actions to all resources, but that resulted in poor performance when there were large numbers of resources being backed up and restored that didn't benefit from the actions of this plugin. Unless there's something this plugin does that your plugin relies on (and I don't think that's the case), then this is unnecessary.

**Review by sseago — APPROVED** [review]

**Review by pranavgaikwad — APPROVED** [review]
